pkg/controllers/subnet/subnet_controller.go

```diff
@@ -332,9 +333,40 @@ func getExistingConditionOfType(conditionType v1alpha1.ConditionType, existingCo
 }
 
 func (r *SubnetReconciler) RestoreReconcile() error {
+	restoreList, err := r.getRestoreList()
+	if err != nil {
+		err = fmt.Errorf("failed to get Subnet restore list: %w", err)
+		return err
+	}
+	var errorList []error
+	for _, key := range restoreList {
+		result, err := r.Reconcile(context.Background(), ctrl.Request{NamespacedName: key})
+		if result.Requeue || err != nil {
+			errorList = append(errorList, fmt.Errorf("failed to restore Subnet %s, error: %w", key, err))
+		}
+	}
+	if len(errorList) > 0 {
+		return errors.Join(errorList...)
+	}
 	return nil
 }
 
+func (r *SubnetReconciler) getRestoreList() ([]types.NamespacedName, error) {
+	restoreList := []types.NamespacedName{}
+	subnetList := &v1alpha1.SubnetList{}
+	if err := r.Client.List(context.TODO(), subnetList); err != nil {
+		return restoreList, err
+	}
+	for _, subnet := range subnetList.Items {
+		// Restore a Subnet if it has NetworkAddresses
+		// Not filter the Subnet in cache as it might be updated
```

Comment:
Can you help explain more why we do not compare with NSX subnet in store?

Reply:
I'm thinking about the case as the following:
Create a Subnet -> Backup -> Update the Subnet -> Create a SubnetPort on the Subnet -> Restore
In such a case, if we do not reconcile the Subnet in store, we will create the SubnetPort before updating the Subnet.

Currently Subnet can switch between dhcpServer and dhcpRelay, which may have impact on how the SubnetPort being created. Though it's not a common usage, I'm thinking about if in the future some other updates are supported for Subnet, we may want to ensure the update happens before SubnetPort creation, i.e. reconcile Subnet to check if it is updated in restore process.